results/cmd/api/main_test.go

```diff
@@ -61,5 +63,35 @@ func setupTestDB(dbName string, t *testing.T) (*sql.DB, error) {
 	if _, err := statement.Exec(); err != nil {
 		t.Fatalf("failed to execute the taskrun table creation statement statement: %v", err)
 	}
-	return db, nil
+	return srv, nil
+}
+
+func TestGetTaskRun(t *testing.T) {
+	// Create a temporay database
+	srv, err := setupTestDB("testdb", t)
+	if err != nil {
+		t.Fatalf("failed to setup db: %v", err)
+	}
+
+	// Connect to fake server and create a new taskrun
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	r, err := srv.CreateTaskRun(ctx, &pb.CreateTaskRunRequest{
+		TaskRun: &pb.TaskRun{
+			ApiVersion: "v1beta1",
+			Metadata: &pb.ObjectMeta{
+				Uid:       "31415926",
+				Name:      "mytaskrun",
+				Namespace: "default",
+			}}})
+	if err != nil {
+		t.Fatalf("could not create taskrun: %v", err)
+	}
+	t.Logf("Created taskrun: %s", r.String())
+
+	// Test if we can find inserted taskrun
+	r, err = srv.GetTaskRun(ctx, &pb.GetTaskRunRequest{Uid: r.GetMetadata().GetUid()})
```

Comment:
We should also validate that we got the response we expected. https://pkg.go.dev/github.com/google/go-cmp/cmp?tab=doc#Diff will be useful here!

Reply:
Equal can not detect TaskRun type, so I used taskrun.String() to convert each one to a string then do the comparison, does it make sense to you?

Reply:
done

Reply:
Reopening (this wasn't done). The `cmp` library should allow you to compare complex structs. In particular, the `Diff` method is probably what you want so that you have a concise output.